test-integration/helpers/ui/test-element.coffee

```diff
@@ -94,10 +97,12 @@ class TestHelper extends TestItemHelper
       defaultWaitTime: 40 * 1000 # TODO: Letting tests define their own wait time is dangerous. tutor-dev takes > 10sec to delete a task-plan
       # defaultWaitTime: 20 * 1000 # 20sec seems to be enough for deployed code but not local
 
+    @test = test
     @_options = _.assign {}, defaultOptions, options
     @_el = {}
 
     commonElements.loadingState = @options.loadingLocator
+    commonElements.self = testElementLocator
```

Comment:
OK. I guess I liked having all the selectors in 1 place and simple constructors. In an HTML comment in this comment I tried writing the helpers without the `.is-loading` but realized it would be crazy to have almost every helper have to define that element (even if `Helpers.User` didn't use it).

<!--
I meant that semantically not every helper describes a screen/element where `.is-loading` could be `true` (`Helpers.User` comes to mind but there _may_ be others). Totally agree that `waitUntilLoaded()` should wait for `self` I was just thinking that `waitUntilLoaded()` could be defined like:

```
constructor: (config) ->
  ...
  # This allows folks to test things before a load finishes, like the grayed out calendar
  @el.loading ?= new TestElement(css: '.is-loading')

waitUntilLoaded: =>
  {loadingState} = @el
  utils.waitUntil(=> loadingState.isPresent.then(isPresent) => !isPresent)
  utils.wait.for(self.isPresent)
```
-->